dali/pipeline/data/tensor_vector.cc

```diff
@@ -12,35 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "tensor_vector.h"
+#include "dali/pipeline/data/tensor_vector.h"
 
 namespace dali {
 
 template <typename Backend>
 TensorVector<Backend>::TensorVector()
     : views_count_(0), tl_(std::make_shared<TensorList<Backend>>()) {}
 
+
 template <typename Backend>
-TensorVector<Backend>::TensorVector(int batch_size)
-    : views_count_(0),
-      tensors_(batch_size, nullptr),
-      tl_(std::make_shared<TensorList<Backend>>(batch_size)) {
-  for (auto &t : tensors_) {
-    t = std::make_shared<Tensor<Backend>>();
-  }
+TensorVector<Backend>::TensorVector(int max_batch_size)
```

Comment:
This is changing the semantics of TensorVector and it's not expected - either based on previous behavor or by analogy to std::vector.

Reply:
Added `batch_size` to TensorList ctor